File: topics.js
const admin = require('firebase-admin');
const fs = require('fs-extra');
const yaml = require('yamljs');

admin.initializeApp({
    credential: admin.credential.cert(require('./credentials.json')),
});
const db = admin.firestore();

const topics = [
    'angular',
    'flutter',
    'cf',
    'firebase',
    'firestore',
    'flutter',
    'rxjs',
    'js',
    'ts'
]

const update = async(topicId) => {

    const json = yaml.load(`topics/${topicId}.yaml`);

    const ref = db.collection('topics').doc(topicId);

    await ref
        .set(json, { merge: true })
        .catch(e => console.log(`'${topicId}' upload failed with error: ${e.details}`));

    console.log(`'${topicId}' is DONE`);
}

for (const topic of topics) {
    update(topic);
}

File: topics.py
import firebase_admin
from firebase_admin import credentials, firestore
import yaml
import asyncio

# Initialize Firebase Admin SDK
cred = credentials.Certificate('./credentials.json')
firebase_admin.initialize_app(cred)
db = firestore.client()

topics = [
    'angular',
    'flutter',
    'cf',
    'firebase',
    'firestore',
    'flutter',
    'rxjs',
    'js',
    'ts'
]

async def update(topic_id):
    with open(f'topics/{topic_id}.yaml', 'r') as file:
        yaml_data = yaml.safe_load(file)

    print(yaml_data)

    ref = db.collection('topics').document(topic_id)

    await ref.set(yaml_data, merge=True)

    print('DONE')

# Loop through topics and update Firestore
async def main():
    tasks = [update(topic) for topic in topics]
    await asyncio.gather(*tasks)

if __name__ == "__main__":
    asyncio.run(main())
